hpx/parallel/util/loop.hpp

```diff
@@ -68,12 +71,194 @@ namespace hpx { namespace parallel { namespace util
     ///////////////////////////////////////////////////////////////////////////
     namespace detail
     {
+
+        //New random access iterator which is used for prefetching
+        //all containers within lambda functions
+        template<typename T>
+        class prefetching_iterator
+        : public std::iterator<std::random_access_iterator_tag, std::size_t>
+        {
+            public:
+
+            using base_iterator = std::vector<std::size_t>::iterator;
```

Comment:
Please don't use `using` to declare type aliases, use `typedef` instead. We still support compilers which don't (fully) support `using`.